fairlearn/metrics/_metric_frame.py

```diff
@@ -382,19 +385,57 @@ def group_max(self) -> Union[Any, pd.Series, pd.DataFrame]:
         whether control features are present, and whether the metric functions
         were specified as a single callable or a dictionary.
 
+        Parameters
+        ----------
+        errors: {'raise', 'coerce'}, default 'raise'
+            if 'raise', then invalid parsing will raise an exception
+            if 'coerce', then invalid parsing will be set as NaN
+
         Returns
         -------
         typing.Any or pandas.Series or pandas.DataFrame
             The maximum value over sensitive features. The exact type
             follows the table in :attr:`.MetricFrame.overall`.
         """
+        if errors not in ("raise", "coerce"):
+            raise ValueError(_INVALID_ERRORS_VALUE_ERROR_MESSAGE)
+
         if not self.control_levels:
-            result = pd.Series(index=self._by_group.columns, dtype='object')
-            for m in result.index:
-                max_val = self._by_group[m].max()
-                result[m] = max_val
+            try:
+                result = pd.Series(index=self._by_group.columns, dtype='object')
+
+                for m in result.index:
+                    max_val = self._by_group[m].max()
+                    result[m] = max_val
+
+            # Catch ValueError that is thrown by pandas, if not all columns in metric frame are scalar
+            except ValueError:
+                warnings.warn(_MF_CONTAINS_NON_SCALAR_WARNING,
+                              SyntaxWarning)
+                if errors == "raise":
+                    raise
+                elif errors == 'coerce':
+                    if not self.control_levels:
+                        # Make all results NaN
+                        result = pd.Series(np.nan, index=self._by_group.columns, dtype='object')
+                        # Fill in the possible columns with their max values
+                        for m in result.index:
+                            if isinstance(self._by_group[m].values[0], np.float64):
```

Comment:
@adrinjalali  should this be `numpy.isscalar()`? Although that looks like it might return `True` for strings.

Reply:
I updated it to np.isscalar(), since this is more readable.